dkg-gadget/src/worker.rs

```diff
@@ -1224,21 +1319,32 @@ where
 			DKGMsgPayload::Offline(..) | DKGMsgPayload::Vote(..) => {
 				let msg = Arc::new(dkg_msg);
 				let async_index = msg.msg.payload.get_async_index();
-				dkg_logging::debug!(target: "dkg_gadget::worker", "Received message for async index {}", async_index);
+				self.logger.debug(format!("Received message for async index {async_index}"));
 				if let Some(Some(rounds)) = self.signing_rounds.read().get(async_index as usize) {
-					dkg_logging::debug!(target: "dkg_gadget::worker", "Message is for signing execution in session {}", rounds.session_id);
+					self.logger.debug(format!(
+						"Message is for signing execution in session {}",
+						rounds.session_id
+					));
 					if rounds.session_id == msg.msg.session_id {
-						dkg_logging::debug!(target: "dkg_gadget::worker", "Message is for this signing execution in session: {}", rounds.session_id);
+						self.logger.debug(format!(
+							"Message is for this signing execution in session: {}",
+							rounds.session_id
+						));
 						if let Err(err) = rounds.deliver_message(msg) {
 							self.handle_dkg_error(DKGError::CriticalError {
 								reason: err.to_string(),
 							})
 						}
 					} else {
-						dkg_logging::debug!(target: "dkg_gadget::worker", "Message is for another signing round: {}", rounds.session_id);
+						self.logger.error(format!(
+							"Message is for another signing round: {}",
+							rounds.session_id
+						));
+						panic!("Message is for another signing round: {}", rounds.session_id)
 					}
 				} else {
-					dkg_logging::trace!(target: "dkg_gadget::worker", "No signing rounds for async index {}", async_index);
+					self.logger.error(format!("No signing rounds for async index {async_index}"));
+					panic!("No signing rounds for async index {async_index}")
```

Comment:
Why are you adding panics to the code? Why is this desirable?

Reply:
Switched to errors